packages/migrate/migrations/page-state/migrate.js

```diff
@@ -0,0 +1,74 @@
+import fs from 'node:fs';
+import { update_pkg } from '../../utils.js';
+
+export function update_pkg_json() {
+	fs.writeFileSync(
+		'package.json',
+		update_pkg_json_content(fs.readFileSync('package.json', 'utf8'))
+	);
+}
+
+/**
+ * @param {string} content
+ */
+export function update_pkg_json_content(content) {
+	return update_pkg(content, [['@sveltejs/kit', '^2.12.0']]);
+}
+
+/**
+ * @param {string} code
+ * @param {{ filename?: string  }} options
+ */
+export function transform_svelte_code(code, options) {
+	// Quick check if nothing to do
+	if (!code.includes('$app/stores')) return code;
+
+	// Check if file is using legacy APIs - if so, we can't migrate since reactive statements would break
+	const lines = code.split('\n');
+	if (lines.some((line) => /^\s*(export let|\$:) /.test(line))) {
+		return code;
+	}
+
+	const import_match = code.match(/import\s*{([^}]+)}\s*from\s*("|')\$app\/stores\2/);
+	if (!import_match) return code; // nothing to do
+
+	const stores = import_match[1].split(',').map((i) => i.trim());
+	let modified = code.replace('$app/stores', '$app/state');
+
+	for (const store of stores) {
+		// if someone uses that they're deep into stores and we better not touch this file
+		if (store === 'getStores') return code;
+
+		const regex = new RegExp(`\\b${store}\\b`, 'g');
```

Comment:
I might be missing something since I'm looking at it from the phone but this doesn't account for

```ts
import { page as page_store } from "$app/stores";
```
right? It will try to substitute the whole `page as page_store` (and probably fail to do so)

Reply:
Yes but IMO that's fine, I haven't seen a single rename anywhere, and even if, it's just that one file that needs manual migration.

As I said in the pr description this is simple text replacement but it works surprisingly well given the unique constraints of dollar prefixes etc

Reply:
Oh yeah...just thought it's probably decently easy to add since is still a determined syntax, but we can probably skip it

Reply:
I did a quick search out of mere curiosity 

https://github.com/search?q=path%3A*.svelte+%2Fimport.%2Bas.%2Bfrom%5Cs%2B%28%5B%27%7C%22%5D%29%5C%24app%5C%2Fstores%2F&type=code

It's objectively rare (200 something instances on millions of files)

Reply:
add support for aliasing